### DISCUSSION TITLE
[drjit] Automatic backend selection

### DISCUSSION THREAD

**Discussion by maxfrei750** [mitsuba-renderer/mitsuba3#646] [Q&A] [answered]
Is there a way to automatically adapt the backend that is used for `drjit` functions, e.g. `dr.zeros`?

Motivation: When writing custom functions, I'd like to use `drjit` data types that match the `mitsuba` variant that the user selected. When creating variables, I can resort to things like `mi.Float`, to automatically get the right data type. Is there something similar for functions like `dr.zeros`?

Or should I create the array of zeros using `numpy` and convert to a vector/tensor using `mitsuba`?

**Comment by maxfrei750** [comment, marked as answer]
The solution was rather simple. Functions like `dr.zeros` expect a data type as first argument. So one can simply pass a dynamic data type like `mi.Float` to them, e.g.:

```
dr.zeros(mi.Float, 3)
```

**Reply by njroussel** [reply]
Hi @maxfrei750 

Indeed, most mitsuba types are just aliases for drjit types. Passing directly the mitsuba type to the few Dr.Jit operations that require it should almost always work :smile:
